lib/processor.js

```diff
@@ -145,6 +156,24 @@ function preprocess(text) {
  * @returns {Message[]} A flattened array of messages with mapped locations.
  */
 function postprocess(messages) {
+
+    /**
+     * Returns messages which are not in the list of expected errors
+     * @param  {Message[]}  msgs A flattened array of messages with mapped locations.
+     * @returns {Message[]}      An array of messages without expected messages.
+     */
+    function getUnexpectedMessages(msgs) {
+        return msgs.filter(function (msg) {
+            var msgIsExpected = false;
+            expectedErrors.forEach(function (expErr) {
```

Comment:
Some kind of map would probably help here to avoid the potential m*n running time.